Index list (#196)
leaningtech/labs: sites/cheerpx/src/content/docs/11-guides/File-System-support.md

});
```

This will mount the specified directory at `/web` in the CheerpX filesystem. To enable listing the directory contents, CheerpX looks for an `index.list` file in the directory and any subdirectories. This file should contain a newline-separated list of all files and subdirectories within the directory.

### Why use an `index.list` file?

> The `index.list` file enables CheerpX to provide directory listing functionality in its WebDevice. It ensures seamless navigation of directories, compatibility with Linux-based tools, and support for navigating directories operations like `os.listdir()`. Since browsers restrict direct access to local files, `index.list` helps simulate a directory structure for virtual environments, making it essential for managing and accessing files effectively.

Reviewer: ```suggestion
The `index.list` file enables CheerpX to provide directory listing functionality in its WebDevice. Since HTTP does not have native support for file listing, `index.list` is essential for managing and accessing files effectively.
```

Please notice how this version is more concise, less redundant and more accurate. On this last point the reference to local files was completely out of place.

### Creating an `index.list` file

To generate an `index.list` in every directory without including it in the directory listing, use the following command:

```bash

> find . -type d -exec sh -c 'cd "{}" && ls > .index.list && mv .index.list index.list' \;

Reviewer: What about `ls {} > .index.list` instead?

Reviewer: Of course `mv` needs to be adapted

});
```

This will mount the specified directory at `/web` in the CheerpX filesystem. To enable listing the directory contents, CheerpX looks for an `index.list` file in the directory and any subdirectories. This file should contain a newline-separated list of all files and subdirectories within the directory.

> ### Why use an `index.list` File?

Reviewer 2: Please do not capitalize words that are not the first word or aren't proper names. Please apply this correction to all the titles and subtitles of this guide.

});
```

### Adding files

You can add files to a DataDevice from JavaScript using the `writeFile` method:

```javascript

> await dataDevice.writeFile("/filename", "file content here");

Reviewer: This capitalization change is certainly not intended
